keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/telemetry/TelemetryListener.kt

```diff
@@ -0,0 +1,38 @@
+package com.netflix.spinnaker.keel.telemetry
+
+import com.netflix.spectator.api.Id
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.keel.info.InstanceIdSupplier
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.slf4j.LoggerFactory
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+@Component
+class TelemetryListener(
+  private val spectator: Registry,
+  private val instanceIdSupplier: InstanceIdSupplier
+) {
+  private val resourceCheckedCounterId = Id.create("keel.resource.checked")
+  private val scope = CoroutineScope(Dispatchers.IO)
+
+  @EventListener(ResourceChecked::class)
+  fun onResourceChecked(event: ResourceChecked) {
+    scope.launch {
+      try {
+        spectator.counter(
+          resourceCheckedCounterId
+            .withTag("resource_name", event.name.value)
+            .withTag("resource_state", event.state.name)
+            .withTag("instance_id", instanceIdSupplier.get())
```

Comment:
Can't remember if I actually need to tag the instance id or if that is done automagically.

Reply:
the instanceId should already be added automatically

Reply:
Yeah, just checked Orca & couldn't see it. Also appears like camel case rather than snake case is the convention for tag names.